<a href="https://colab.research.google.com/github/CharlesNasci/Paradigmas/blob/master/Copy_of_USJT_2020_LINGPROG_aula03_pratica_ifs_e_loops.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

![logo_usjt.png](https://www.usjt.br/app/themes/mobister/dist/images/sao-judas-universidade.png)

# Paradigmas de Linguagens de Programação
- **Professor**: Sergio Bonato <antonio.bonato@saojudas.br>

# Atividade03 - Python Básico - Ifs, Loops e Arquivos

**Conteúdo:**
- comparações, condicionais
- laços for e while
- arquivos

## Autor
- **Aluno:**

Obs: as tarefas devem ser individuais.

## Exercícios

**1** Faça um programa para a leitura de duas notas parciais de um aluno. O programa deve calcular a média alcançada por aluno e apresentar:
- A mensagem "Aprovado", se a média alcançada for maior ou igual a sete;
- A mensagem "Reprovado", se a média for menor do que sete;
- A mensagem "Aprovado com Distinção", se a média for igual a dez.

In [ ]:
#solução
nota1 = float(input('Insira a primeira nota do aluno '))
nota2 = float(input('Insira a segunda nota do aluno '))

media = nota1 + nota2 / 2
if media >= 7:
  print('Aprovado')

elif media == 10:
  print('Aprovado com Distinção')
  
else:
  print('Reprovado')


Insira a primeira nota do aluno 10
Insira a segunda nota do aluno 8
Aprovado


**2** Faça um programa que leia três números e mostre-os em ordem decrescente.

In [ ]:
#solução
n1 = int(input('insira um numero:'))
n2 = int(input('insira um numero:'))
n3 = int(input('insira um numero:'))

print(n1,n2,n3)

if (n1 < n2):
  aux = n3
  n3 = n2
  n2 = aux

if (n2 > n1):
  aux = n2
  n2 = n1
  n1 = aux

if(n3 > n2):
  aux = n3
  n3 = n2
  n2 = aux

print (n1,n2,n3)

insira um numero1
insira um numero2
insira um numero3
1 2 3
3 2 1


**3** As Organizações Tabajara resolveram dar um aumento de salário aos seus colaboradores e lhe contrataram para desenvolver o programa que calculará os reajustes.

Faça um programa que recebe o salário de um colaborador e o reajuste segundo o seguinte critério, baseado no salário atual:
- salários até R\$ 280,00 (incluindo) : aumento de 20%
- salários entre R\$ 280,00 e R\$ 700,00 : aumento de 15%
- salários entre R\$ 700,00 e R\$ 1500,00 : aumento de 10%
- salários de R\$ 1500,00 em diante : aumento de 5% 

Após o aumento ser realizado, informe na tela:
- o salário antes do reajuste;
- o percentual de aumento aplicado;
- o valor do aumento;
- o novo salário, após o aumento.

In [ ]:
#solução
salario = float(input('Qual o salario do colaborador? '))

if (salario <= 280):
  percentual = 20
elif (salario <=700):
  percentual = 15
elif (salario <=1500):
  salario = 10
else:
  percentual = 5

print ('Salario original: R$ ', salario)
print('Percentual',percentual,'%')

percentual = percentual/100.0
aumento = percentual * salario
nSalario = salario + aumento

print('Aumento: R$ ', aumento)
print('Novo salario: R$ ', nSalario)



Qual o salario do colaborador? 100
Salario original: R$  100.0
Percentual 20 %
Aumento: R$  20.0
Novo salario: R$  120.0


**4** Faça um Programa que leia um número e exiba o dia correspondente da semana. (1-Domingo, 2- Segunda, etc.), se digitar outro valor deve aparecer valor inválido.

In [ ]:
#solução
semana = int(input('Digite um numero de 1 a 7: '))

if semana == 1:
  print('1 -> Domingo')
elif semana == 2:
  print('2 -> Segunda feira')
elif semana == 3:
  print('3 -> Terça feira')
elif semana == 4:
  print('4 -> Quarta-feira')
elif semana == 5:
  print('5 -> Quinta feira')
elif semana == 6:
  print('6 -> Sexta-feira')
elif semana == 7:
  print('7 -> sabado')
else:
  print('valor invalido')
  
print (semana)

Digite um numero de 1 a 7: 10
valor invalido
10


**5** Faça um Programa que peça os 3 lados de um triângulo. O programa deverá informar se os valores podem ser um triângulo. Indique, caso os lados formem um triângulo, se o mesmo é: equilátero, isósceles ou escaleno.

Dicas:
- Três lados formam um triângulo quando a soma de quaisquer dois lados for maior que o terceiro;
- Triângulo Equilátero: três lados iguais;
- Triângulo Isósceles: quaisquer dois lados iguais;
- Triângulo Escaleno: três lados diferentes;


In [ ]:
#solução
l1 = int(input('Digita o valor de um dos lados do triangulo: '))
l2 = int(input('Digita o valor de um dos lados do triangulo: '))
l3 = int(input('Digita o valor de um dos lados do triangulo: '))

if l1>=l2+l3 or l2>=l3+l1 or l3>=l1+l2:
  print('nao pode ser um triangulo')
elif l1 == l2 == l3:
  print ('Triangulo equilatero')
elif (l1 == l2) or (l1 == l3) or (l2 == l3):
  print('Triangulo Isosceles')
else:
  print ('Triangulo escaleno')


Digita o valor de um dos lados do triangulo: 5
Digita o valor de um dos lados do triangulo: 5
Digita o valor de um dos lados do triangulo: 5
Triangulo equilatero


**6** Faça um programa que calcule as raízes de uma equação do segundo grau, na forma `ax2 + bx + c`. O programa deverá pedir os valores de `a`, `b` e `c` e fazer as consistências, informando ao usuário nas seguintes situações:

- Se o usuário informar o valor de A igual a zero, a equação não é do segundo grau e o programa não deve fazer pedir os demais valores, sendo encerrado;
- Se o delta calculado for negativo, a equação não possui raizes reais. Informe ao usuário e encerre o programa;
- Se o delta calculado for igual a zero a equação possui apenas uma raiz real; informe-a ao usuário;
- Se o delta for positivo, a equação possui duas raiz reais; informe-as ao usuário.

In [ ]:
#solução

a = float(input('Insira um valor '))
b = float(input('insira um valor '))
c = float(input('insira um valor '))

delta = 0
raiz1 = 0
raiz2 = 0

if a == 0: 
  print('A equecação nao é de segundo grau')

else:
  print('Calculando delta...')
  delta = (((b**2) - (4*a*c))) 

if delta == 0:
  print('A equação possui apenas uma raiz que é: ')
  raiz1 = ((-b)+(delta**(1/2))/(2*a))
  print(raiz1)
else:
  raiz1 = ((-b)+(delta**(1/2))/(2*a))
  raiz2 = ((-b)-(delta**(1/2))/(2*a))
  print('A equação possui duas raizes que sao: ', raiz1, 'e', raiz2)


Insira um valor 4
insira um valor 2
insira um valor 8
Calculando delta...
A equação possui duas raizes que sao:  (-2+1.3919410907075054j) e (-2-1.3919410907075054j)


**7** Faça um programa que faça 5 perguntas para uma pessoa sobre um crime. As perguntas são:
- "Telefonou para a vítima?"
- "Esteve no local do crime?"
- "Mora perto da vítima?"
- "Devia para a vítima?"
- "Já trabalhou com a vítima?" 

O programa deve no final emitir uma classificação sobre a participação da pessoa no crime. Se a pessoa responder positivamente a 2 questões ela deve ser classificada como "Suspeita", entre 3 e 4 como "Cúmplice" e 5 como "Assassino". Caso contrário, ele será classificado como "Inocente".

In [ ]:
#solução
print('que comece os jogos')
print('--------------------')

res = []

res.append (input('Telefonou para a vitima? 1/Sim ou 0/Nao):'))
res.append (input('Esteve no local do crime? 1/Sim ou 0/Nao: '))
res.append (input('Mora perto da vitima? 1/Sim ou 0/Nao: '))
res.append (input('Devia para a vitima? 1/Sim ou 0/Nao: '))
res.append (input('Ja trabalhou com a vitima? 1/Sim ou 0/Nao'))

soma = 0
for i in res:
  soma += int(i)
if (soma < 2):
  print('Inocente')
elif (soma == 2):
  print('Suspeito')
elif(3 <= soma <=4):
  print('Cumplice')
elif(soma == 5):
  print('Assassino')

que comece os jogos
--------------------
Telefonou para a vitima? 1/Sim ou 0/Nao):1
Esteve no local do crime? 1/Sim ou 0/Nao: 0
Mora perto da vitima? 1/Sim ou 0/Nao: 1
Devia para a vitima? 1/Sim ou 0/Nao: 0
Ja trabalhou com a vitima? 1/Sim ou 0/Nao1
Cumplice


**8** Faça um programa que peça uma nota, entre zero e dez. Mostre uma mensagem caso o valor seja inválido e continue pedindo até que o usuário informe um valor válido.

In [ ]:
#solução
valor = int(input('Insira um valor de 0-10 '))

while (valor > 0) or (valor < 10):
  valor = int(input('Insira um valor de 0-10 '))
  print('valor invalido')




Insira um valor de 0-10 10
Insira um valor de 0-10 5
valor invalido
Insira um valor de 0-10 11
valor invalido


**9** Faça um programa que leia 5 números e informe a soma e a média dos números.

In [ ]:
#solução
n1 = int(input('Insira o 1º numero: '))
n2 = int(input('Insira o 2º numero: '))
n3 = int(input('Insira o 3º numero: '))
n4 = int(input('Insira o 4º numero: '))
n5 = int(input('Insira o 5º numero: '))

soma = n1 + n2 + n3 + n4 + n5
media = soma /5

print ('O valor da toma = ', soma,'e sua media = ',media)

Insira o 1º numero10
Insira o 2º numero203
Insira o 3º numero4
Insira o 4º numero8
Insira o 5º numero6
O valor da toma =  231 e sua media =  46.2


**10** Faça um programa que imprima na tela apenas os números ímpares entre 1 e 50.

In [ ]:
#solução
for i in  range(1,50,2):
  print (i)

1
3
5
7
9
11
13
15
17
19
21
23
25
27
29
31
33
35
37
39
41
43
45
47
49


**11** Desenvolva um gerador de tabuada, capaz de gerar a tabuada de qualquer número inteiro entre 1 a 10. O usuário deve informar de qual numero ele deseja ver a tabuada. A saída deve ser conforme o exemplo abaixo:

`
Tabuada do 5:
5 X 1 = 5
5 X 2 = 10
...
5 X 10 = 50
`

In [ ]:
#solução
num = int(input('Qual tabuada voce deseja?, Digite um numero entre 1 e 10: '))

print('Tabuada do numero', num)
for i in range (1,11):
  print(num, 'X', i, '= ',num * i)


Qual tabuada voce deseja?, Digite um numero entre 1 e 104
Tabuada do numero 4
4 X 1 =  4
4 X 2 =  8
4 X 3 =  12
4 X 4 =  16
4 X 5 =  20
4 X 6 =  24
4 X 7 =  28
4 X 8 =  32
4 X 9 =  36
4 X 10 =  40


**12** A série de Fibonacci é formada pela seqüência 1,1,2,3,5,8,13,21,34,55,... Faça um programa capaz de gerar a série até o n−ésimo termo.

In [ ]:
#solução
n = int(input('até que termo deseja encontrar a sequencia? '))
ultimo = 1
penultimo = 1 
if (n ==1) or (n ==2):
  print('1')
else:
  count = 3
  while count <= n:
    termo = ultimo + penultimo
    penultimo = ultimo
    ultimo = termo
    count += 1
    print(termo)



até que termo deseja encontrar a sequencia? 10
2
3
5
8
13
21
34
55


**13** Faça um programa que calcule o fatorial de um número inteiro fornecido pelo usuário. Ex.: 5!=5.4.3.2.1=120

In [ ]:
#solução
n = int(input('Digite o valor de um numero: '))

fatorial = 1

for i in range (2,n +1):
  fatorial = fatorial * i
  print("%d! = %d" %(n, fatorial))


Digite o valor de um numero: 5
5! = 2
5! = 6
5! = 24
5! = 120


**14** Faça um programa que receba o valor de uma dívida e mostre uma tabela com os seguintes dados: valor da dívida, valor dos juros, quantidade de parcelas e valor da parcela.

Os juros e a quantidade de parcelas seguem a tabela abaixo:

|Quantidade de Parcelas|% de Juros sobre o valor inicial da dívida|
|:-|:-|
|1       |0
|3       |10
|6       |15
|9       |20
|12      |25

Exemplo de saída do programa:

|Valor da Dívida|Valor dos Juros|Quantidade de Parcelas|Valor da Parcela|
|:-|:-|:-|:-|
|1.000,00|0|1|1.000,00
|1.100,00|100|3|366,00
|1.150,00|150|6|191,67


In [ ]:
#solução
divida = float(input('Insira o valor da divida: '))
parcela = float(input('Insira o valor de parcela: '))

print('\n' * 5)
print('Valor da divida: ', end=' ')
print('valor do juros: ' , end=' ')
print('Quantidade de parcelas: ', end = ' ')
print('Valor da parcela: ')

for i in range(5):
  if parcela == 1:
    juros = 1
    valor_juros = 0
  elif parcela == 4:
    parcela = 3
    juros = 1.10
  elif parcela == 7 or parcela == 6:
    parcela = 6
    juros = 1.15
  elif parcela == 10 or parcela == 9:
    parcela = 9
    juros = 1.20
  elif parcela == 13 or parcela == 12:
    parcela = 12
    juros = 1.25

valor_juros = divida * (juros -1)
divida_com_juros = divida * juros
valor_parcela = divida_com_juros / parcela

print('R$', round(divida_com_juros,2), end= '                ')
print(round(valor_juros,2), end = '           ')
print(parcela, end = '                      ')
print('R$', round(valor_parcela,2))
parcela += 3


Insira o valor da divida: 50
Insira o valor de parcela: 2






Valor da divida:  valor do juros:  Quantidade de parcelas:  Valor da parcela: 
R$ 55.0                5.0           2.0                      R$ 27.5


**15** A ACME Inc., uma empresa de 500 funcionários, está tendo problemas de espaço em disco no seu servidor de arquivos. Para tentar resolver este problema, o Administrador de Rede precisa saber qual o espaço ocupado pelos usuários, e identificar os usuários com maior espaço ocupado. Através de um programa, baixado da Internet, ele conseguiu gerar o seguinte arquivo, chamado "usuarios.txt":

```
alexandre       456123789
anderson        1245698456
antonio         123456456
carlos          91257581
cesar           987458
rosemary        789456125
```

Neste arquivo, o nome do usuário possui 15 caracteres. A partir deste arquivo, você deve criar um programa que gere um relatório, chamado "relatório.txt", no seguinte formato:

```
ACME Inc.   Uso do espaço em disco pelos usuários
-------------------------------------------------
Nr.  Usuário        Espaço utilizado     % do uso
---- -------------- ---------------- ------------
1    alexandre             434,99 MB       16,85%
2    anderson             1187,99 MB       46,02%
3    antonio               117,73 MB        4,56%
4    carlos                 87,03 MB        3,37%
5    cesar                   0,94 MB        0,04%
6    rosemary              752,88 MB       29,16%

Espaço total ocupado: 2581,57 MB
Espaço médio ocupado: 430,26 MB
```


O arquivo de entrada deve ser lido uma única vez, e os dados armazenados em memória, caso sejam necessários, de forma a agilizar a execução do programa. A conversão da espaço ocupado em disco, de bytes para megabytes deverá ser feita através de uma função separada, que será chamada pelo programa principal. O cálculo do percentual de uso também deverá ser feito através de uma função, que será chamada pelo programa principal.


In [ ]:
def toMB(tamanhoEmBytes):
   tamanhoEmBytes = float(tamanhoEmBytes)
   return (float(tamanhoEmBytes/(1024*1024)))
   
def percentualPorUsuario(lista, total):
   percentual = (lista[3]/total)*100
   #percentual = "{0:.2f}".format(percentual)
   lista.insert((len(cadaUsuario)),percentual)

def espacoMedioOcupado(lista, total):
   media = 0
   elementos = len(lista)
   media = (total)/(elementos+1)
   #media = "{0:.2f}".format(media)
   return media

#MAIN
usuarios = []
posicao = 1
total = media = 0
with open ("usuarios.txt","r") as arquivo:
   valor = 0
   for linha in arquivo:
      usuarios.append(linha.split()) 

   for cadaUsuario in usuarios:
      cadaUsuario.insert(0,posicao)
      valor = toMB(float(cadaUsuario[2]))
      total += valor
      cadaUsuario.insert((len(cadaUsuario)),valor)
      posicao+=1

   for cadaUsuario in usuarios:
      percentualPorUsuario(cadaUsuario, total)

media = espacoMedioOcupado(cadaUsuario,total)

with open ("relatorio.txt","w") as arquivo:
   arquivo.write("ACME Inc.               Uso do espaço em disco pelos usuários.\n")
   arquivo.write("--------------------------------------------------------------\n")
   arquivo.write("Nr.\tUsuário        \tEspaço utilizado\t% do uso\n\n")

   for cadaUsuario in usuarios:
      percentagem="{0:.2f}".format(round(cadaUsuario[3],2))
      arquivo.write(str(cadaUsuario[0])+'\t'+"{:<15}".format(cadaUsuario[1])+'\t'+"{:<16}".format(percentagem)+'MB'+'\t'+"{0:.2f}".format(cadaUsuario[4])+'%'+'\n')

   arquivo.write('\n\nEspaço Total Ocupado: ' + "{0:.2f}".format(total) + ' MB')
   arquivo.write('\n\nEspaço médio Ocupado: ' + "{0:.2f}".format(media) + ' MB')
   arquivo.close()

with open ("relatorio.txt","r") as arquivo:
   print(arquivo.read())

FileNotFoundError: ignored